### PR TITLE
chore: refactor fund tests to test/lib

### PR DESCRIPTION
**Pull request by ruyadorno**
Refactored `npm fund` tests to use new `test/lib/` unit tests structure.

ref: npm/statusboard#151

### CODE REVIEW

File: lib/fund.js [hunk 1]
```diff
@@ -131,8 +122,10 @@ async function openFundingUrl ({ path, tree, spec, fundingSourceNumber }) {
   if (!funding) {
```
Comment (isaacs) [inline lib/fund.js]: Could this whole section be written to use a `const`?  Maybe something like:

```js
const { funding } = retrievePackageMetadata() || 
  await (pacote.manifest(arg, npm.flatOptions).catch(() => ({}))
```

File: tap-snapshots/test-lib-fund.js-TAP.test.js [hunk 2]
```diff
@@ -5,7 +5,11 @@
  * Make sure to inspect the output below.  Do not ignore changes!
  */
 'use strict'
-exports[`test/tap/fund.js TAP fund containing multi-level nested deps with no funding > should omit dependencies with no funding declared 1`] = `
+exports[`test/lib/fund.js TAP fund a package with type and multiple sources > should print name only 1`] = `
+
```
Comment (isaacs) [inline tap-snapshots/test-lib-fund.js-TAP.test.js:12]: Curious, this says "should print name only", but it's not printing anything?

**Review by isaacs — COMMENTED** [review]
One eyebrow-raise (might just be a confusing test name?) and a suggestion to tighten up code in one spot to replace a `let` with a `const`.  LGTM otherwise, and if those comments aren't valid, feel free to ignore them :)

File: lib/fund.js (hunk 1 above)
Comment (ruyadorno) [reply lib/fund.js]: that looks neat! yep, will change it 😊

File: tap-snapshots/test-lib-fund.js-TAP.test.js (hunk 2 above)
Comment (ruyadorno) [reply tap-snapshots/test-lib-fund.js-TAP.test.js:12]: oooooops, will review what's going on 😬

**Comment by ruyadorno** [comment]
should be good to go now ✅

**Comment by isaacs** [comment]
Landed on 68a16d6dbc796f8f527bc28643a111c3ca2a9b1d, thanks!

**Pull request closed without merging**